Sources/TuistKit/Commands/GenerateCommand.swift

```diff
@@ -54,16 +57,33 @@ class GenerateCommand: NSObject, Command {
         projectOnlyArgument = subParser.add(option: "--project-only",
                                             kind: Bool.self,
                                             usage: "Only generate the local project (without generating its dependencies).")
+        
+        carthageProjectsArgument = subParser.add(option: "--carthage-projects",
```

Comment:
I would also like to add in the option for automatically fetching and building carthage dependencies.

Tuist can:
  - perform `carthage checkout --use-ssh` if the Carthage/Checkouts directory doesn't exist
  - generate all projects (including those in Carthage/Checkouts)
  - execute `carthage build --platform iOS --configuration Debug` <- Infer the platform from the targets which have carthage dependencies 

TBC ^